src/Account.sol

```diff
@@ -747,30 +797,39 @@ contract Account is IAccount, Auth, OpsReady {
     }
 
     /*//////////////////////////////////////////////////////////////
-                   GELATO CONDITIONAL ORDER HANDLING
+                       EXECUTE CONDITIONAL ORDER
     //////////////////////////////////////////////////////////////*/
 
     /// @inheritdoc IAccount
     function executeConditionalOrder(uint256 _conditionalOrderId)
         external
         override
+        nonReentrant
         isAccountExecutionEnabled
-        onlyOps
     {
-        // store conditional order in memory
+        // store conditional order object in memory
         ConditionalOrder memory conditionalOrder =
             getConditionalOrder(_conditionalOrderId);
 
+        // verify conditional order is ready for execution
+        /// @dev it is understood this is a duplicate check if the executor is Gelato
+        if (!_validConditionalOrder(_conditionalOrderId)) {
```

Comment:
**Suggestion:** Within `_validConditionalOrder` I think it would be good to have the following check:
```solidity
        if (conditionalOrder.marketKey == bytes32("")) {
            return false;
        }
```